From 49ee3573417075982a98d231bf0ee31fd69efac8 Mon Sep 17 00:00:00 2001
From: Jaska Uimonen <jaska.uimonen@intel.com>
Date: Tue, 24 Jul 2012 14:33:03 +0300
Subject: [PATCH 13/13] release: added no-verify to suppress whitespace errors
 in git

---
 obs-export.sh |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/obs-export.sh b/obs-export.sh
index 43c66d6..f7e0260 100755
--- a/obs-export.sh
+++ b/obs-export.sh
@@ -157,7 +157,7 @@ if [ "$MODE" = "gerrit" -a "$RELEASE" = "yes" ]; then
     git branch $branch $UPSTREAM_BASE && \
         git checkout $branch && \
         git add packaging && \
-        git commit -m "release: added packaging for gerrit." packaging && \
+        git commit -n -m "release: added packaging for gerrit." packaging && \
         echo "* $(date '+%a %b %d %H:%M:%S %Z %Y') $AUTHOR - $VERSION" \
             > $chlog && \
         echo "- release: releasing $VERSION..." >> $chlog && \
-- 
1.7.7.6

